test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -741,24 +748,60 @@ class ExtProcIntegrationTest : public HttpIntegrationTest,
     return response;
   }
 
-  void serverReceiveHeaderDuplexStreamed(ProcessingRequest& header_request) {
-    EXPECT_TRUE(grpc_upstreams_[0]->waitForHttpConnection(*dispatcher_, processor_connection_));
-    EXPECT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
-    EXPECT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, header_request));
-    EXPECT_TRUE(header_request.has_request_headers());
+  void serverReceiveHeaderDuplexStreamed(ProcessingRequest& header, bool first_message = true,
+                                         bool response = false) {
+    if (first_message) {
+      EXPECT_TRUE(grpc_upstreams_[0]->waitForHttpConnection(*dispatcher_, processor_connection_));
+      EXPECT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+    }
+    EXPECT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, header));
+    if (!response) {
```

Comment:
remove the `!` and flip the then/else clauses.

Reply:
done